cranelift/codegen/src/isa/zkasm/inst/emit.rs

```diff
@@ -442,6 +443,26 @@ impl MachInstEmit for Inst {
                     sink,
                 );
             }
+            &Inst::UExtend {
+                rd,
+                rs,
+                ty_in,
+                ty_out,
+            } => {
+                // TODO: support other types cast
+                if ty_in == I32 && ty_out == I64 {
+                    let rs = allocs.next(rs);
+                    let rd = allocs.next_writable(rd);
+
+                    debug_assert_eq!(rs, e0());
+                    put_string("4294967296n => B\n", sink);
+                    put_string("0 => D\n", sink);
+                    // Intentionnally don't put E % B to C. If E % B != 0 something goes wrong.
+                    put_string("0 => C\n", sink);
+                    put_string("${E / B} => A\n", sink);
+                    put_string("E:ARITH\n", sink);
+                }
```

Comment:
Let's also add `unimplemented!` outside of this case to make sure we don't silently ignore all other conversions.

Reply:
I'm not shure we should do this, because as I understand this function is called for casts like ``u8->u16`` (and for such casts we should silently do nothing, and idk why, cranelift generate such casts).  I suggest file issue to rework casts, because this PR is more about "make ``u32->u64`` cast working ASAP".

Reply:
The danger of silently doing nothing here is that if there are casts that are expected to have some effects, we will never learn about them - we should at least consider adding a log message here saying that the cast occurred.